wagtail/core/fields.py

```diff
@@ -91,7 +94,17 @@ def __init__(self, block_types, **kwargs):
         self.stream_block.set_meta_options(block_opts)
 
     def get_internal_type(self):
-        return "TextField"
+        return "JSONField" if self.use_json_field else "TextField"
+
+    def get_lookup(self, lookup_name):
+        if self.use_json_field:
+            return models.JSONField().get_lookup(lookup_name)
```

Comment:
Instead of calling `models.JSONField()` everywhere, might be worth creating a `@cached_property`?

Reply:
Ah yes, the `encoder` and `decoder` could probably affect it (I doubt it in this case, though). Anyway, I've set the `encoder` to `DjangoJSONEncoder` in all `JSONField()` calls as that's what `StreamField` normally uses.